rest-openehr/src/main/java/org/ehrbase/rest/openehr/OpenehrDefinitionQueryController.java

```diff
@@ -71,89 +73,114 @@ public OpenehrDefinitionQueryController(QueryService queryService) {
      * @param qualifiedQueryName
      * @return
      */
-    @EhrbaseAuthorization(permission = EhrbasePermission.EHRBASE_QUERY_READ)
-    @RequestMapping(
-            value = {"/{qualified_query_name}", ""},
-            method = RequestMethod.GET)
     @Override
+    @GetMapping(value = {"/{qualified_query_name}", ""})
+    @EhrbaseAuthorization(permission = EhrbasePermission.EHRBASE_QUERY_READ)
     public ResponseEntity<QueryDefinitionListResponseData> getStoredQueryList(
             @RequestHeader(value = ACCEPT, required = false) String accept,
             @PathVariable(value = "qualified_query_name", required = false) String qualifiedQueryName) {
 
-        log.debug("getStoredQueryList invoked with the following input: " + qualifiedQueryName);
+        logger.debug("getStoredQueryList invoked with the following input: {}", qualifiedQueryName);
 
         QueryDefinitionListResponseData responseData =
                 new QueryDefinitionListResponseData(queryService.retrieveStoredQueries(qualifiedQueryName));
         return ResponseEntity.ok(responseData);
     }
 
-    @EhrbaseAuthorization(permission = EhrbasePermission.EHRBASE_QUERY_READ)
-    @RequestMapping(
-            value = {"/{qualified_query_name}/{version}"},
-            method = RequestMethod.GET) //
     @Override
+    @GetMapping(value = {"/{qualified_query_name}/{version}"})
+    @EhrbaseAuthorization(permission = EhrbasePermission.EHRBASE_QUERY_READ)
     public ResponseEntity<QueryDefinitionResponseData> getStoredQueryVersion(
             @RequestHeader(value = ACCEPT, required = false) String accept,
             @PathVariable(value = "qualified_query_name") String qualifiedQueryName,
             @PathVariable(value = "version") Optional<String> version) {
 
-        log.debug("getStoredQueryVersion invoked with the following input: " + qualifiedQueryName + ", version:"
-                + version);
+        logger.debug(
+                "getStoredQueryVersion invoked with the following input: {}, version:{}", qualifiedQueryName, version);
 
         QueryDefinitionResponseData queryDefinitionResponseData = new QueryDefinitionResponseData(
                 queryService.retrieveStoredQuery(qualifiedQueryName, version.isPresent() ? version.get() : null));
 
         return ResponseEntity.ok(queryDefinitionResponseData);
     }
 
+    @Override
     @EhrbaseAuthorization(permission = EhrbasePermission.EHRBASE_QUERY_CREATE)
-    @RequestMapping(
+    @PutMapping(
             value = {"/{qualified_query_name}/{version}", "/{qualified_query_name}"},
-            method = RequestMethod.PUT)
-    @Override
+            produces = {TEXT_PLAIN_VALUE, APPLICATION_JSON_VALUE})
```

Comment:
On second thought, TEXT_PLAIN and APPLICATION_JSON is consumed, but only APPLICATION_JSON_VALUE (or XML?) is produced.
(I believe if supported XML but would have failed due to the Gson parsing)

Reply:
done